app/Main.hs

```diff
@@ -26,24 +32,70 @@ liveConf mgr = ExchangeConf mgr Nothing Live
 sandboxConf :: Manager -> ExchangeConf
 sandboxConf mgr = ExchangeConf mgr Nothing Sandbox
 
--- | One round of waiting and requesting the next set of data
-keepGettingWindowsWithDelay :: Int -> World -> IO World
-keepGettingWindowsWithDelay delayDuration prevWindow = do
-  threadDelay delayDuration
-  getNextWindow prevWindow 
+{----- ALGORITHM AND LOOP -----}
+
+-- | Trading algorithm 
+makeDecision :: World -> LookingTo -> Decision
+makeDecision (World _ (Window (EMA short, EMA long) _)) (LookingTo Buy) =
+  if short > long
+  then Decision Buy
+  else Hold
+makeDecision (World _ (Window (EMA short, EMA long) _))  (LookingTo Sell) =
+  if short < long -- threshold here to sell sooner and be risk averse
+  then Decision Sell
+  else Hold
+
+-- | Fill in with actual API calls,
+-- | but for now just print to see what bot does
+executeDecision :: (Decision, World) -> LookingTo -> IO LookingTo
+executeDecision (Hold, (World _ window)) lookingTo= do
+  putStrLn ("Held with market price at: " ++ show (unPrice window))
+  return lookingTo
+executeDecision (Decision Sell, (World _ window)) _ = do
+  putStrLn ("Sold at price: " ++ show (unPrice window))
+  return (LookingTo Buy)
+executeDecision (Decision Buy, (World _ window)) _ = do
+  putStrLn ("Bought at price: " ++ show (unPrice window))
+  return (LookingTo Sell)
+
+  -- Variables
+  -- percentage for limit order: 1/200
+  -- time per window: 3 minutes
+  -- short length: 10
+  -- long length: 30
+  -- threshold for putting the sell order
+
+-- | Tie everything together 
+makeAndExecuteDecisions :: Stream (Of World) IO () -> Stream (Of LookingTo) (StateT LookingTo IO) ()
+makeAndExecuteDecisions worlds = do
+  worldEither <- liftIO $ S.next worlds
+  case worldEither of
+    Left _ -> error "yikes my world ended"
```

Comment:
This can be made safe by replacing `error ...` here with `pure ()`. Alternately you can have type `Stream (Of World) IO () -> Stream (Of LookingTo) (StateT LookingTo IO) Void`, using the `Data.Void` package, as a way of signaling that these streams will never return. (Void is an uninhabitable type.)